proposals/async-method-builders.md

```diff
@@ -46,23 +46,70 @@ namespace System.Runtime.CompilerServices
     /// Indicates the type of the async method builder that should be used by a language compiler to
     /// build the attributed method.
     /// </summary>
-    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Struct | AttributeTargets.Interface | AttributeTargets.Method | AttributeTargets.Module, Inherited = false, AllowMultiple = false)]
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Struct | AttributeTargets.Interface | AttributeTargets.Method | AttributeTargets.Module, Inherited = false, AllowMultiple = true)]
```

Comment:
Think we should include `AttributeTargets.Assembly` here. Yes the compiler team is typically focused on modules because that is the unit of emit that we have to deal with. 99% of users though don't understand the difference between assembly and module and they are all used to assembly level attributes. Think we should support that and ask users to write it.

Reply:
For skiplocalsinit, the [attribute ](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.skiplocalsinitattribute?view=net-5.0) is allowed on modules, but not assemblies. 
I think our other attribute-based features did the same. We should probably stick to that approach.

On the other hand, should we allow `Property` (accessors could use async lambdas or local functions), `Event` (ditto) and `Constructor`?

Reply:
I'm proposing that we should allow `Property`, `Event` and `Constructor`, but not `Assembly`. Also added a tracking issue in open questions.